prov/gni/src/gnix_msg.c

```diff
@@ -135,36 +234,14 @@ static int __comp_eager_msg_w_data(void *data)
 	ep = req->gnix_ep;
 	assert(ep != NULL);
 
-	cq = ep->send_cq;
-	assert(cq != NULL);
-
-	cq_len = _gnix_cq_add_event(cq,
-				    req->user_context,
-				    FI_SEND | FI_MSG,
-				    0,
-				    0,
-				    0,
-				    0);
-	if (cq_len != FI_SUCCESS)  {
-		GNIX_WARN((FI_LOG_CQ | FI_LOG_EP_DATA),
-			  "_gnix_cq_add_event returned %d\n",
-			   cq_len);
-		ret = (int)cq_len; /* ugh */
-	}
-
-	if (ep->send_cntr) {
-		ret = _gnix_cntr_inc(ep->send_cntr);
-		if (ret != FI_SUCCESS)
-			GNIX_WARN(FI_LOG_CQ,
-			  "_gnix_cntr_inc returned %d\n",
-			   ret);
-	}
+	__gnix_send_completion(ep, req);
 
 	atomic_dec(&req->vc->outstanding_tx_reqs);
+	_gnix_nic_tx_free(req->gnix_ep->nic, tdesc);
```

Comment:
TXs were previously freed in gnix_nic.c after the request callback function returned.  This mod moves freeing of the TX into the request callbacks (__comp_*() functions).  This was done so the request callback functions can choose to reuse a TX for multi part requests.  Specifically, we now reuse a TX after completing rendezvous GETs.